### PR TITLE
[React] Error Message component

### PR DESCRIPTION
**Pull request by clairesunstudio**
error

### CODE REVIEW

**Comment by avrilpearl** [comment]
@clairesunstudio  --> do you want to change the font color on different status?

screen shot 2018-08-13 at 2 35 47 pm

File: react/src/components/atoms/forms/ErrorMessage/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  message: PropTypes.string.isRequired,
+  /** Validation status */
+  status: PropTypes.oneOf(['error, success'])
+};
+
+ErrorMessage.defaultProps = {
+  status: 'error'
+};
+
+export default ErrorMessage;
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/ErrorMessage/index.js]: you need to add this to the index.js component export file...

File: react/src/components/atoms/forms/ErrorMessage/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import SvgInputSuccess from '../../icons/SvgInputSuccess';
+import './style.css';
+
+const ErrorMessage = ({inputId, message, status}) => {
+  const isSuccessful = status === 'success';
+  return(
+    <div
+      htmlFor={inputId}
+      aria-labelledby={inputId}
+      className={`ma__error-msg has-error ${isSuccessful ? 'ma__error-msg--success' : ''}`}
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/ErrorMessage/index.js:15]: should probably add something in style.scss for rendering the message text (font) green in a success state.

**Comment by clairesunstudio** [comment]
The styling didn't work because mayflower-react is still pulling from mayflower package css and the styles were competing. Added temporary overrides to make it work now and will address that after the share scss pr gets in

**Review by smurrayatwork — CHANGES_REQUESTED** [review]
The ErrorMessage component has a bunch of lint things that need fixing.

File: react/src/components/atoms/forms/ErrorMessage/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  )
+};
+
+ErrorMessage.propTypes = {
+  /** The ID of the corresponding input field */
+  inputId: PropTypes.string.isRequired,
+  /** The help text for the corresponding input field */
+  message: PropTypes.string.isRequired,
+  /** Validation status */
+  status: PropTypes.oneOf(['error, success'])
```
Comment (smurrayatwork) [inline react/src/components/atoms/forms/ErrorMessage/index.js]: I think these should be two different strings?

File: react/src/components/atoms/forms/ErrorMessage/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import SvgInputError from '../../icons/SvgInputError';
+import SvgInputSuccess from '../../icons/SvgInputSuccess';
+import './style.css';
+
+const ErrorMessage = ({inputId, message, status}) => {
```
Comment (smurrayatwork) [inline react/src/components/atoms/forms/ErrorMessage/index.js]: Should the inputId actually be the id attribute for the div being rendered?

**Review by smurrayatwork — CHANGES_REQUESTED** [review]

File: react/src/components/atoms/forms/ErrorMessage/index.js (hunk 4 above)
Comment (clairesunstudio) [reply react/src/components/atoms/forms/ErrorMessage/index.js]: no that should be the id for `<input>` that the error message is tied to

File: react/src/components/atoms/forms/ErrorMessage/index.js (hunk 3 above)
Comment (clairesunstudio) [reply react/src/components/atoms/forms/ErrorMessage/index.js]: good catch

Comment (clairesunstudio) [reply react/src/components/atoms/forms/ErrorMessage/index.js]: fixed this and the lint errors --> a77ab0d

**Review by smurrayatwork — APPROVED** [review]

File: react/src/components/atoms/forms/ErrorMessage/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import SvgInputError from '../../icons/SvgInputError';
+import SvgInputSuccess from '../../icons/SvgInputSuccess';
+import './style.css';
+
+const ErrorMessage = ({ inputId, message, status }) => {
```
Comment (avrilpearl) [inline react/src/components/atoms/forms/ErrorMessage/index.js]: is there a reason you called the out here vs. as just props? - we are not doing this anywhere else so curious...

**Review by avrilpearl — APPROVED** [review]

**Pull request merged**